source/logic.cpp

```diff
@@ -175,10 +175,17 @@ namespace Logic {
   bool FreeFairies      = false;
   bool FairyPond        = false;
   bool BombchuDrop      = false;
+  bool AmmoCanDrop      = false;
 
   bool BuyBombchus5     = false;
   bool BuyBombchus10    = false;
   bool BuyBombchus20    = false;
+  bool BuySeed          = false;
+  bool BuyArrow         = false;
+  bool BuyBomb          = false;
+  bool BuyGPotion       = false;
+  bool BuyBPotion       = false;
+  bool MagicRefill      = false;
```

Comment:
I just realized- You forgot to add these variables to LogicReset(). All you have to do is put them there in the same order they're in here and set them to false.